netket/nn/linear.py

```diff
@@ -207,6 +208,99 @@ def __call__(self, inputs: Array) -> Array:
         return y
 
 
+class DenseSymm(Module):
+    permutations: Any
+    alpha: Union[float, int]
+    use_bias: bool = True
+    dtype: Any = jnp.float64
+    precision: Any = None
+    kernel_init: Callable[[PRNGKey, Shape, Dtype], Array] = default_kernel_init
+    bias_init: Callable[[PRNGKey, Shape, Dtype], Array] = zeros
+
+    def setup(self):
+        perms = self.permutations()
+
+        self.n_symm, self.n_sites = perms.shape
+        self.features = int(self.alpha * self.n_sites / self.n_symm)
+        self.n_hidden = self.features * self.n_symm
+        if self.alpha > 0 and self.features == 0:
+            raise ValueError(
+                f"RBMSymm: alpha={self.alpha} is too small "
+                f"for {self.n_symm} permutations, alpha ≥ {self.n_symm / self.n_sites} is needed."
+            )
+
+        # symmetrization tensor, maps Wsymm to full W
+        # note that this should be a sparse object, once jax supports them
+        def symmetrizer_ijkl(i, j, k, l):
+            jsymm = np.floor_divide(j, self.n_symm)
+            cond_k = k == jnp.asarray(perms)[j % self.n_symm, i]
+            cond_l = l == jsymm
+            return jnp.asarray(jnp.logical_and(cond_k, cond_l), dtype=int)
+
+        self.symmetrizer = jnp.array(
+            np.fromfunction(
+                symmetrizer_ijkl,
+                shape=(self.n_sites, self.n_hidden, self.n_sites, self.features),
+                dtype=int,
+            )
+        )
+
+    def matmul(self, inputs: Array, kernel: Array) -> Array:
```

Comment:
maybe mark it as internal?
```suggestion
    def _matmul(self, inputs: Array, kernel: Array) -> Array:
```